0.0.1/agent.py

```diff
@@ -435,35 +457,88 @@ def execute_swap(self):
             self.recommend_token_allocations_to_swap_for_stablecoins()
 
         # Get quotes for both USDC and USDT
-        usdc_quotes = get_usdc_quotes(self.recommended_tokens)
-        usdt_quotes = get_usdt_quotes(self.recommended_tokens)
+        amount_in = self.recommended_tokens.get("NEAR", 0)
+        self.env.add_system_log(
+            f"Fetching quotes to swap {amount_in} Near for USDC/USDT",
+            logging.DEBUG
+        )
 
-        # Choose the highest value
-        if usdc_quotes.usd_value > usdt_quotes.usd_value:
-            best_quote = usdc_quotes
-        else:
-            best_quote = usdt_quotes
+        quotes = await self._client.get_stablecoin_quotes("nep141:wrap.near", amount_in, self.near_account_id)
 
-        # TODO create an intent payload with best_quote
+        # Choose best quote between USDC and USDT based on minimum amount out with slippage taken into account
+        best_quote = None
+        if quotes.get("USDC") and quotes.get("USDT"):
+            usdc_amount = Decimal(quotes["USDC"].quote.get("minAmountOut", 0))
+            usdt_amount = Decimal(quotes["USDT"].quote.get("minAmountOut", 0))
+            best_quote = quotes["USDC"] if usdc_amount > usdt_amount else quotes["USDT"]
+        elif quotes.get("USDC"):
+            best_quote = quotes["USDC"]
+        elif quotes.get("USDT"):
+            best_quote = quotes["USDT"]
 
```

Comment:
could we put this in its own util function `get_best_quote`?

Reply:
logged in a new issue